sage_modabvar/torsion_subgroup.py

```diff
@@ -332,25 +377,123 @@ def divisor_of_order(self):
             4383
 
         """
+        try:
+            return self._divisor_of_order
+        except:
+            pass
+
         A = self.abelian_variety()
         N = A.level()
+
         if A.dimension() == 0:
-            return ZZ(1)
+            self._divisor_of_order = 1
+            return self._divisor_of_order
+
+        # return the order of the cuspidal subgroup in the J0(p) case
+        if A.is_J0() and N.is_prime():
+            self._divisor_of_order = QQ((A.level()-1)/12).numerator()
+            return self._divisor_of_order
+
+        # The elliptic curve case
+        if A.dimension() == 1:
+            self._divisor_of_order = A.elliptic_curve().torsion_order()
+            return self._divisor_of_order
 
         # The J1(p) case
         if A.is_J1() and N.is_prime():
             epsilons = [epsilon for epsilon in DirichletGroup(N)
                         if not epsilon.is_trivial() and epsilon.is_even()]
             bernoullis = [epsilon.bernoulli(2) for epsilon in epsilons]
-            return ZZ(N/(2**(N-3))*prod(bernoullis))
+            self._divisor_of_order = ZZ(N/(2**(N-3))*prod(bernoullis))
+            return self._divisor_of_order
 
+        # The Gamma0 case
         if all(is_Gamma0(G) for G in A.groups()):
-            R = A.rational_cusp_subgroup()
-            return R.order()
+            self._divisor_of_order = A.rational_cusp_subgroup().order()
+            return self._divisor_of_order
+
+        # Unhandled case
+        self._divisor_of_order = ZZ(1)
+        return self._divisor_of_order
+
+    def multiple_of_order(self, maxp=None, proof=True):
+        """
+        Return a multiple of the order.
+
+        INPUT:
+
+        - ``proof`` -- a boolean (default: True)
+
+        The computational of the rational torsion order of J1(p) is conjectural
+        and will only be used if proof=False. See Section 6.2.3 of [CES2003]_.
+
+        EXAMPLES::
+
+            sage: from sage_modabvar import J0, J1
+            sage: J = J1(11); J
+            Abelian variety J1(11) of dimension 1
+            sage: J.rational_torsion_subgroup().multiple_of_order()
+            5
+
+            sage: J = J0(17)
+            sage: J.rational_torsion_subgroup().order()
+            4
+
+        This is an example where proof=False leads to a better bound and better
+        performance. ::
+
+            sage: J = J1(23)
+            sage: J.rational_torsion_subgroup().multiple_of_order() # long time (2s)
```

Comment:
The developer guide said to do it for anything longer than a second. 

http://doc.sagemath.org/html/en/developer/doctesting.html#run-long-tests